src/language-js/print/jsx.js

```diff
@@ -480,13 +480,15 @@ function printJsxAttribute(path, options, print) {
     if (isStringLiteral(node.value)) {
       const raw = rawText(node.value);
       // Unescape all quotes so we get an accurate preferred quote
-      let final = raw.replace(/&apos;/g, "'").replace(/&quot;/g, '"');
-      const quote = getPreferredQuote(
+      let final = raw
+        .replace(/&apos;/g, "'")
+        .replace(/&quot;/g, '"')
+        .slice(1, -1); // remove enclosing quotes
```

Comment:
I think this should slice first.

Reply:
Makes sense - it happend to work because the attributes' enclosing quotes might never be escaped.